src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -250,19 +260,38 @@ public void run() {
     driver.run();
   }
 
-  private void launchNode(SchedulerDriver driver, Offer offer,
-        String nodeName, List<String> taskNames, String executorName) {
+  private boolean launchNode(SchedulerDriver driver, Offer offer,
+        String nodeName, List<String> taskTypes, String executorName) {
     log.info(String.format("Launching node of type %s with tasks %s", nodeName,
-        taskNames.toString()));
+        taskTypes.toString()));
     String taskIdName = String.format("%s.%s.%d", nodeName, executorName,
         System.currentTimeMillis());
     List<Resource> resources = getExecutorResources();
     ExecutorInfo executorInfo = createExecutor(taskIdName, nodeName, executorName, resources);
     List<TaskInfo> tasks = new ArrayList<>();
-    for (String taskName : taskNames) {
-      List<Resource> taskResources = getTaskResources(taskName);
+    for (String taskType : taskTypes) {
```

Comment:
Let's break this into a separate method, please since the method is getting large.